diff -Nur a/Modules/Platform/GNU.cmake b0/Modules/Platform/GNU.cmake
--- a/Modules/Platform/GNU.cmake	2020-06-28 14:25:40.908491990 -0000
+++ b0/Modules/Platform/GNU.cmake	2020-06-28 14:42:31.185715729 -0000
@@ -26,13 +26,8 @@
   # checking the platform every time.  This option is advanced enough
   # that only package maintainers should need to adjust it.  They are
   # capable of providing a setting on the command line.
-  if(EXISTS "/etc/debian_version")
-    set(CMAKE_INSTALL_SO_NO_EXE 1 CACHE INTERNAL
-      "Install .so files without execute permission.")
-  else()
-    set(CMAKE_INSTALL_SO_NO_EXE 0 CACHE INTERNAL
-      "Install .so files without execute permission.")
-  endif()
+  set(CMAKE_INSTALL_SO_NO_EXE 0 CACHE INTERNAL
+    "Install .so files without execute permission.")
 endif()
 
 set(CMAKE_LIBRARY_ARCHITECTURE_REGEX "[a-z0-9_]+(-[a-z0-9_]+)?-gnu[a-z0-9_]*")
diff -Nur a/Modules/Platform/Linux.cmake b0/Modules/Platform/Linux.cmake
--- a/Modules/Platform/Linux.cmake	2020-06-28 14:25:40.908491990 -0000
+++ b0/Modules/Platform/Linux.cmake	2020-06-28 14:39:30.501204554 -0000
@@ -37,13 +37,8 @@
   # checking the platform every time.  This option is advanced enough
   # that only package maintainers should need to adjust it.  They are
   # capable of providing a setting on the command line.
-  if(EXISTS "/etc/debian_version")
-    set(CMAKE_INSTALL_SO_NO_EXE 1 CACHE INTERNAL
-      "Install .so files without execute permission.")
-  else()
-    set(CMAKE_INSTALL_SO_NO_EXE 0 CACHE INTERNAL
-      "Install .so files without execute permission.")
-  endif()
+  set(CMAKE_INSTALL_SO_NO_EXE 0 CACHE INTERNAL
+    "Install .so files without execute permission.")
 endif()
 
 # Match multiarch library directory names.
@@ -51,9 +46,3 @@
 
 include(Platform/UnixPaths)
 
-# Debian has lib32 and lib64 paths only for compatibility so they should not be
-# searched.
-if(NOT CMAKE_CROSSCOMPILING AND EXISTS "/etc/debian_version")
-  set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS FALSE)
-  set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS FALSE)
-endif()
