diff -Nur a/Modules/GNUInstallDirs.cmake b1/Modules/GNUInstallDirs.cmake
--- a/Modules/GNUInstallDirs.cmake	2020-06-28 14:48:24.754543337 -0000
+++ b1/Modules/GNUInstallDirs.cmake	2020-06-28 14:43:54.035784260 -0000
@@ -227,24 +227,11 @@
       "Please enable at least one language before including GNUInstallDirs.")
   endif()
   if(CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU)$"
-      AND NOT CMAKE_CROSSCOMPILING
-      AND NOT EXISTS "/etc/arch-release")
-    if (EXISTS "/etc/debian_version") # is this a debian system ?
-      if(CMAKE_LIBRARY_ARCHITECTURE)
-        if("${CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
-          set(_LIBDIR_DEFAULT "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
-        endif()
-        if(DEFINED _GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX
-            AND "${_GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
-          set(__LAST_LIBDIR_DEFAULT "lib/${CMAKE_LIBRARY_ARCHITECTURE}")
-        endif()
-      endif()
-    else() # not debian, rely on CMAKE_SIZEOF_VOID_P:
-      if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-        set(_LIBDIR_DEFAULT "lib64")
-        if(DEFINED _GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX)
-          set(__LAST_LIBDIR_DEFAULT "lib64")
-        endif()
+      AND NOT CMAKE_CROSSCOMPILING)
+    if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
+      set(_LIBDIR_DEFAULT "lib64")
+      if(DEFINED _GNUInstallDirs_LAST_CMAKE_INSTALL_PREFIX)
+        set(__LAST_LIBDIR_DEFAULT "lib64")
       endif()
     endif()
   endif()
