diff -Nur a/src/uapi/linux/linux/wireguard.h b/src/uapi/linux/linux/wireguard.h
--- a/src/uapi/linux/linux/wireguard.h	2020-07-08 23:38:51.000000000 +0800
+++ b/src/uapi/linux/linux/wireguard.h	2020-07-13 20:13:39.561457741 +0800
@@ -157,6 +157,7 @@
 	WGDEVICE_A_LISTEN_PORT,
+	WGDEVICE_A_BIND_ADDR,
 	WGDEVICE_A_FWMARK,
 	WGDEVICE_A_PEERS,
 	__WGDEVICE_A_LAST
 };
 #define WGDEVICE_A_MAX (__WGDEVICE_A_LAST - 1)
